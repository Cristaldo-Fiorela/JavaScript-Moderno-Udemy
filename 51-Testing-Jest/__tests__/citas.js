import Citas from '../js/classes/Citas';

describe('Crear la clase de citas', () => {
    const citas = new Citas;
    const id = Date.now();

    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Fiore',
            telefono: '123124',
            fecha: '10-12-2023',
            hora:'8.00p.m',
            sintomas: 'Solo duerme'
        }

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });

    test('Actualizar cita', () => {
        const citaActualizada = {
            id,
            mascota: 'Donna',
            propietario: 'Fiore',
            telefono: '123124',
            fecha: '10-12-2023',
            hora:'8.00p.m',
            sintomas: 'Solo duerme'
        };

        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar citas', () => {
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    })
});