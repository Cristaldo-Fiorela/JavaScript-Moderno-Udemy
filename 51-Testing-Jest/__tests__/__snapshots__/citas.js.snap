// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crear la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "8.00p.m",
      "id": 1683324803312,
      "mascota": "Donna",
      "propietario": "Fiore",
      "sintomas": "Solo duerme",
      "telefono": "123124",
    },
  ],
}
`;

exports[`Crear la clase de citas Agregar una nueva cita 1`] = `
Citas {
  "citas": [
    {
      "fecha": "10-12-2023",
      "hora": "8.00p.m",
      "id": 1683324803312,
      "mascota": "Hook",
      "propietario": "Fiore",
      "sintomas": "Solo duerme",
      "telefono": "123124",
    },
  ],
}
`;

exports[`Crear la clase de citas Eliminar citas 1`] = `
Citas {
  "citas": [],
}
`;
